core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala

```diff
@@ -245,6 +245,7 @@ object ResourceProfile extends Logging {
   // Executor resources
   val CORES = "cores"
   val MEMORY = "memory"
+  val OFFHEAP_MEM = "offHeap"
   val OVERHEAD_MEM = "memoryOverhead"
   val PYSPARK_MEM = "pyspark.memory"
 
```

Comment:
Any hints about how the test look like? I'm unfamiliar with where this field will be used.

Reply:
Sure, in ResourceProfileSuite, lets create a test that has every type of resource in allSupportedExecutorResources (including off heap) and then say 2 custom resources. Then just call getCustomExecutorResources to ensure that you only get the 2 custom resources back.  We may also have it check the size of allSupportedExecutorResources == 5, that way if someone modifies allSupportedExecutorResources but doesn't update the test it will fail.

Could you also add a big comment   after "// Executor resources" that say make sure to update allSupportedExecutorResources